File: lobster-input-android/app/src/main/java/com/lobster/input/keyboard/pinyin/HotwordDictionaryBridge.kt
package com.lobster.input.keyboard.pinyin

import android.content.Context
import com.lobster.input.core.network.ApiConfig

/**
 * 将 App 主进程同步的热词写入键盘用户词典。
 * MainViewModel 在热词增删改后调用 [persistHotwords];引擎启动时 [importInto] 导入调频。
 */
object HotwordDictionaryBridge {

    const val PREFS_KEY = "ime_hotwords_cache"

    fun persistHotwords(context: Context, words: List<String>) {
        val cleaned = words.map { it.trim() }.filter { it.isNotEmpty() }.distinct()
        context.getSharedPreferences(ApiConfig.PREFS_NAME, Context.MODE_PRIVATE)
            .edit()
            .putString(PREFS_KEY, cleaned.joinToString("\n"))
            .apply()
    }

    fun readHotwords(context: Context): List<String> {
        val raw = context.getSharedPreferences(ApiConfig.PREFS_NAME, Context.MODE_PRIVATE)
            .getString(PREFS_KEY, null) ?: return emptyList()
        return raw.split('\n').map { it.trim() }.filter { it.isNotEmpty() }
    }

    fun importInto(userDict: UserDictionary, context: Context, baseBoost: Int = 8) {
        for (word in readHotwords(context)) {
            repeat(baseBoost) { userDict.learn(word) }
        }
    }
}

File: lobster-input-ios/LobsterKeyboard/Pinyin/HotwordDictionaryBridge.swift
import Foundation

/// 将 App 主进程同步的热词写入键盘用户词典(App Group,与 [UserDictionary] 同域)。
enum HotwordDictionaryBridge {

    static let prefsKey = "ime_hotwords_cache"

    private static var defaults: UserDefaults {
        UserDefaults(suiteName: APIConfig.appGroupID) ?? .standard
    }

    static func persistHotwords(_ words: [String]) {
        let cleaned = words
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
            .filter { !$0.isEmpty }
        let unique = Array(Set(cleaned))
        defaults.set(unique.joined(separator: "\n"), forKey: prefsKey)
    }

    static func readHotwords() -> [String] {
        guard let raw = defaults.string(forKey: prefsKey), !raw.isEmpty else { return [] }
        return raw.split(separator: "\n").map { String($0).trimmingCharacters(in: .whitespacesAndNewlines) }.filter { !$0.isEmpty }
    }

    static func importInto(_ userDict: UserDictionary, baseBoost: Int = 8) {
        for word in readHotwords() {
            for _ in 0..<baseBoost { userDict.learn(word) }
        }
    }
}
